src/components/common/header/AuthenticatedHeader.js

```diff
@@ -1,23 +1,30 @@
-import React, { Fragment, useEffect } from 'react';
-import { Link } from 'react-router-dom';
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { logoutAccount } from '@Redux/actions/authActions';
+import { connect } from 'react-redux';
+import { Link, withRouter } from 'react-router-dom';
 import Materialize from 'materialize-css';
 
-export default () => {
+const AuthenticatedHeader = ({ logout, history }) => {
   useEffect(() => {
     const dropDownElems = document.querySelectorAll('.dropdown-trigger');
     const sideNavElems = document.querySelectorAll('.sidenav');
     Materialize.Sidenav.init(sideNavElems);
     Materialize.Dropdown.init(dropDownElems);
   });
 
+  const logoutUser = () => {
+    logout(logoutAccount(history));
```

Comment:
I don't understand this..🤔

Reply:
@line 17 I'm dispatching logout Action, in the parameter I passed in `history`, so that I can call history.push when it has dispatched successfully. 

**Code snippet of logout Action**
```
export const logoutAccount = history => async (dispatch) => {
  try {
    await API_SERVICE.post('/auth/logout');
    dispatch(logout());
    localStorage.removeItem('jwtToken');
    return history.push('/');
  } catch (error) {
    const { data: { errors } } = error.response;
    const message = Object.values(errors)[0];
    return dispatch(authFailed(message));
  }
};
```